src/main/java/bookstore/entity/cart/CartItem.java

```diff
@@ -0,0 +1,41 @@
+package bookstore.entity.cart;
+
+import bookstore.entity.book.Book;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Getter
+@Setter
+@ToString
+@Table(name = "cart_items")
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
+public class CartItem {
```

Comment:
Why do you need Soft Delete approach here? 

You don't have Soft Delete approach on your shopping cart. Therefore, there will eventually pop up a problem which is after deleting shopping cart you will be left with lots cart items without referred Shopping Cart floating in your database and consuming memory for nothing. Also, your cart items will still have a reference to User, meanwhile not actually being in their shopping cart, which is strange.

The other thing is with your functionality you never actually delete Cart items after deletion of Shopping Cart. Thus, not only they will be floating in your db, but still remain available for interacting through endpoints or services without being in any shopping cart. In this case the soft delete approach isnt even used.